File: app/src/main/java/com/fed/notes/Note.java
package com.fed.notes;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import java.util.Date;
import java.util.UUID;

/**
 * Created by f on 05.05.2017.
 */

@Entity(tableName = "notes")
public class Note {
    @PrimaryKey
    @NonNull
    public UUID id;
    public String title;
    public String description;
    public Date date;

    public Note() {
        this(UUID.randomUUID());
    }

    public Note(UUID id) {
        this.id = id;
        date = new Date();
    }

//    public UUID getId() {
//        return id;
//    }
//
//    public void setId(UUID id) {
//        id = id;
//    }
//
//    public String getTitle() {
//        return title;
//    }
//
//    public void setTitle(String title) {
//        title = title;
//    }
//
//    public String getDescription() {
//        return description;
//    }
//
//    public void setDescription(String description) {
//        description = description;
//    }
//
//    public Date getDate() {
//        return date;
//    }
//
//    public void setDate(long date) {
//        date = new Date(date);
//    }

    public String getPhotoFilename() {
        return "IMG_" + id.toString() + ".jpg";
    }
}

File: app/src/main/java/com/fed/notes/database/Entity/Note.kt
package com.fed.notes.database.Entity

import android.arch.persistence.room.Entity
import android.arch.persistence.room.PrimaryKey

import java.util.Date
import java.util.UUID

@Entity(tableName = "notes")
class Note private constructor(@field:PrimaryKey
                               var id: UUID,
                               var title: String? = null,
                               var description: String? = null,
                               var date: Date = Date()) {

    constructor() : this(UUID.randomUUID())

    fun getPhotoFilename() = "IMG_" + id.toString() + ".jpg"
}